src/pydocstyle/checker.py

```diff
@@ -199,7 +199,9 @@ def check_no_blank_before(self, function, docstring):  # def
             if blanks_before_count != 0:
                 yield violations.D201(blanks_before_count)
             if not all(blanks_after) and blanks_after_count != 0:
-                yield violations.D202(blanks_after_count)
+                def_follows = after.split("\n")[2].lstrip().startswith('def')
```

Comment:
To allow both classes and functions you could use, and avoid including variables that have a name starting with _def_:
```python
re(r"\s+(?:class|def)\s").match(after)
```
This follows how python's re module is used in this file, where `re` is an alias for `re.compile`. It would be good to add a test for the `class` case too.

The check docstring could use updating to reflect the allowance of a single line if it is before a class/function definition.